iphone/Classes/ImageLoader.m

```diff
@@ -504,7 +504,30 @@ -(ImageCacheEntry *)entryForKey:(NSURL *)url
 			{
 				scaleUp = YES;
 			}
-			UIImage * resultImage = [UIImage imageWithContentsOfFile:path];
+			UIImage *resultImage = nil;
+			NSRange range = [urlString rangeOfString:@".app"];
+			NSString *imageArg = nil;
+			if (range.location != NSNotFound) {
+				imageArg = [urlString substringFromIndex:range.location+5];
+			}
+			if (imageArg != nil) {
+				unsigned char digest[CC_SHA1_DIGEST_LENGTH];
+				NSData *stringBytes = [imageArg dataUsingEncoding: NSUTF8StringEncoding];
```

Comment:
Close, so close.

`imageArg` originates from a URL which means it's URL encoded. Spaces become `%20` and such. We need `imageArg` to be URL decoded first.

NSString has a handy dandy function `stringByReplacingPercentEscapesUsingEncoding` that decodes the string, however this function has been deprecated in favor of `stringByRemovingPercentEncoding`.

I may be mistaken, but I think we're going to need to do an `#if` and use the appropriate method:

```objc
#if IS_XCODE_7
				imageArg = [imageArg stringByRemovingPercentEncoding];
#else
				imageArg = [imageArg stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
#endif
```

I've tested the code above by placing it before line 515 and it is able to decode and sha correctly.